sample/src/main/java/io/getstream/chat/example/MainActivity.java

```diff
@@ -77,6 +88,54 @@ public void onError(String errMsg, int errCode) {
                 Log.e(TAG, String.format("Failed to establish websocket connection. Code %d message %s", errCode, errMsg));
             }
         });
+
+        // Configure and adding notification options for notifications
+        StreamNotificationOptions notificationOptions = new StreamNotificationOptions();
+
+        // Set custom intent provider for receiving message and events from firebase and WS
+        notificationOptions.setNotificationIntentProvider(
+                new NotificationIntentProvider() {
+                    @Override
+                    public PendingIntent getIntentForFirebaseMessage(@NonNull Context context, @NonNull RemoteMessage remoteMessage) {
+                        Map<String, String> payload = remoteMessage.getData();
+                        Intent intent = new Intent(context, ChannelActivity.class);
+                        intent.putExtra(EXTRA_CHANNEL_TYPE, payload.get(StreamNotificationsManager.CHANNEL_TYPE_KEY));
+                        intent.putExtra(EXTRA_CHANNEL_ID, payload.get(StreamNotificationsManager.CHANNEL_ID_KEY));
+                        return PendingIntent.getActivity(context, 999,
+                                intent, PendingIntent.FLAG_UPDATE_CURRENT
+                        );
+                    }
+
+                    @Override
+                    public PendingIntent getIntentForWebSocketEvent(@NonNull Context context, @NonNull Event event) {
+                        Intent intent = new Intent(context, ChannelActivity.class);
+                        intent.putExtra(EXTRA_CHANNEL_TYPE, StringUtility.getChannelTypeFromCid(event.getCid()));
+                        intent.putExtra(EXTRA_CHANNEL_ID, StringUtility.getChannelIdFromCid(event.getCid()));
+                        return PendingIntent.getActivity(context, 999,
+                                intent, PendingIntent.FLAG_UPDATE_CURRENT);
+                    }
+                }
+        );
+
+        // Device register listener
+        DeviceRegisteredListener onDeviceRegistered = new DeviceRegisteredListener() {
+            @Override
+            public void onDeviceRegisteredSuccess() {
+                // Device successfully registered on server
+                Log.i(TAG, "Device registered successfully");
+            }
+
+            @Override
+            public void onDeviceRegisteredError(String errorMessage, Integer errorCode) {
+                // Some problem with registration
+                Log.e(TAG, "onDeviceRegisteredError: " + errorMessage + " Code: " + errorCode);
+            }
+        };
+
+        StreamChat.setNotificationsManager(new StreamNotificationsManager(notificationOptions, onDeviceRegistered));
```

Comment:
Let's move notification config to application class.
Some developers might copy and paste sample code into their activities. And that's a one time config, there is no need to config notification every time activity is launched/created.

Application is for one time global configs.
Activities is for specific screens. MainActivity in this case shouldn't care about notifications at all.